### DISCUSSION TITLE
Issue creating basic countdown with special format

### DISCUSSION THREAD

**Discussion by ynsbleng** [moment/moment#6413] [Q&A] [answered]
When formatting a date the rounding must be taken into account, otherwise the result will be inaccurate and so confusing, for instance I am trying to create a basic countdown with a special format DD [DAYS] HH:mm:ss, the backend server is returning a number of seconds until the end of the event, let's say: 27327 seconds, which is 07:35:27, for this I convert the value to milliseconds then use the moment.utc to convert it to a moment object, the final code is as simple as:

`moment.utc(27327 * 1000).format("DD [DAYS] HH:mm:ss")`

Surprisingly this code is returning: 01 DAYS 07:35:27 when it should be: 00 DAYS 07:35:27, so either I am missing something in your doc or there is truly an issue here, either way I think there should be a way in the formatting function to, at least, control the rounding behavior, or create a separate variable name to describe the number of days left so instead of DD it could be DL or whathever.

Also, even if we try to use moment duration then it will be necessary to do the formatting manually, as it's not providing any way to make a the format except for converting it to milliseconds and using moment utc again, which will result in the same behavior, as the duration object will have:

days: 0
hours: 7
milliseconds: 0
minutes: 35
months: 0
seconds: 27
years: 0

but the moment utc using the number of milliseconds of this duration will sadly have that 1 day issue again, so probably implementing a format function for the duration will be the best approach.

Something like that could do the trick:

```
moment.duration.fn.format = function(format){
    return format.replace("{s}", this.seconds()).replace("{m}", this.minutes()).replace("{h}", this.hours()).replace("{d}", this.days()).replace("{ss}", this.seconds().toString().padStart(2, "0")).replace("{mm}", this.minutes().toString().padStart(2, "0")).replace("{hh}", this.hours().toString().padStart(2, "0")).replace("{dd}", this.days().toString().padStart(2, "0"));
}
```

And could be used like:

`moment.duration({ seconds: 27327 }).format("{d} DAYS {hh}:{mm}:{ss}")`

Thanks,

**Comment by tajagn01** [comment]
Hi! I’d like to contribute a fix or enhancement for this — could you please point me to the file where this code lives?

**Comment by q-o-ap** [comment]
#5674

**Comment by mattjohnsonpint** [comment, marked as answer]
Thanks for the clear example! `moment.utc(milliseconds)` creates an instant near the Unix epoch, so `DD` means January's calendar day, not elapsed days. A countdown is a duration. Moment core does not provide arbitrary duration formatting, but [moment-duration-format](https://github.com/jsmreese/moment-duration-format) does: `moment.duration(27327, 'seconds').format('DD [DAYS] HH:mm:ss')`.
